### PR TITLE
CRASM-3008 - Add new null use case alerts to VS Dashboard

### PR DESCRIPTION
**Pull request by ameliav**
#  #

## 🗣 Description ##
Handle Error Messaging and VS Dashboard display for two specific Null Data use cases
CRASM-3008
CRASM-3267

## 💭 Motivation and context ##
As a user, I want to receive clear and informative error messages when data is missing or unavailable, so that I understand what went wrong and can decide on the next steps.

## 🧪 Testing ##
Use below acceptance criteria to determine how to test:
- If null data is encountered (but there is assets owned and hosts scanned):
- Display an error message that explains the issue in plain language.
- Display "--" instead of a 0 for assets owned or assets scanned if the organization is over 2 weeks old.
- Return standardized error messages.

## 📷 Screenshots (if appropriate) ##

If there are assets owned and hosts scanned and nothing else as a standard user and then as a admin or global view user

image
image

If there are assets owned and no hosts scanned and nothing else as a standard user and then as a admin or global view user

image
image

If there's even one vulnerability, the alert will disappear and the normal dashboard will appear.
image

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [ ] *All* future TODOs are captured in issues, which are referenced in code comments.
- [ ] All relevant type-of-change labels have been added.
- [ ] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [ ] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [ ] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.
- [ ] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).
- [ ] Create a pre-release (necessary if and only if the pre-release version was bumped).

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release (necessary if and only if the version was bumped).

### CODE REVIEW

**Review by hawkishpolicy — DISMISSED** [review]
LGTM. Moving the email links is clean.

File: frontend/src/components/Dashboard/CustomAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Alert, AlertTitle, Typography, Link } from '@mui/material';
+
+interface CustomAlertProps {
+  headerMsg?: string;
```
Comment (cduhn17) [inline frontend/src/components/Dashboard/CustomAlert.tsx:5]: Looks close. If you add the small test for the null alert, a guard in VSDashboardGate when no org id is available, and a unit test for isEmptyAfterScans. These are low-effort and reduce the risk of UX regressions around the new branch.

**Review by cduhn17 — COMMENTED** [review]
See comment

File: frontend/src/components/Dashboard/CustomAlert.tsx (hunk 1 above)
Comment (ameliav) [reply frontend/src/components/Dashboard/CustomAlert.tsx:5]: As discussed in the PR meeting, will add CRASM-3267 here first- and then add the tests for both cases with your suggestions, request re-reviews, and present Friday.

Comment (ameliav) [reply frontend/src/components/Dashboard/CustomAlert.tsx:5]: Please see below commits per your requests on testing
- Tests to check if data is empty, null data scenarios, or present: 3426684090ce5452b68e32077095781767c55cdc
- Tests to check from the api call including when no org id is available: bb9feb25f8845d0b90c9b8d29f5c2668e1b73bbf
- Unit tests for isEmptyAfterScans and related utility functions: 9341f80ef9da54c385a25629c6e5325700b474c4

**Review by jsalinasnttdata — APPROVED** [review]
The code changes look good. Please note I have not tested the changes locally.

File: frontend/src/components/Dashboard/CustomAlert.tsx (hunk 1 above)
Comment (cduhn17) [reply frontend/src/components/Dashboard/CustomAlert.tsx:5]: Thanks for the changes

**Review by cduhn17 — APPROVED** [review]
LGTM - thanks for the changes per comments
